src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1176,6 +1178,10 @@ protected String checkJavaVersion(final PrintStream logger, String javaCommand,
         return null;
     }
 
+    private double requiredJavaVersion() {
+        return Double.valueOf("1." + JDK_MAJOR_VERSION);
```

Comment:
Based on Jenkins.getInstance().getVersion() we have 3 possible outcomes: 
before 1.610 then java 6
before 2.5x: java 7
afterwards java 8 (will happen soon; confirmed on the IRC channel).